### ISSUE TITLE
SF: allow "project deploy cancel" to be executed outside of an SFDX project

### ISSUE THREAD

**Issue by hudec117** [forcedotcom/cli#2224]
### Summary
The project deploy cancel command (replacing force:mdapi:deploy:cancel) requires an SFDX project to cancel a deployment. The previous command did not have this requirement and the transition is causing difficulties with CI/CD since our directory structure does not conform to an SFDX project.

### Steps To Reproduce

1. Open command prompt outside of an SF/SFDX project and try to execute `sf project deploy cancel`

### Expected result
Attempted to cancel the on-going deployment in the org set as global default.

### Actual result
```
RequiresProjectError: This command is required to run from within a Salesforce project directory.
Code: RequiresProjectError
```

### System Information
Shell: cmd.exe

```json
{
  "cliVersion": "@salesforce/cli/1.83.0",
  "architecture": "win32-x64",
  "nodeVersion": "node-v18.15.0",
  "osVersion": "Windows_NT 10.0.19045",
  "shell": "C:\\Program Files\\Git\\usr\\bin\\bash.exe",
  "rootPath": "C:\\Program Files\\sfdx\\client\\sf",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.3.0 (core)",
    "@oclif/plugin-commands 2.2.15 (core)",
    "@oclif/plugin-help 5.2.9 (core)",
    "@oclif/plugin-not-found 2.3.24 (core)",
    "@oclif/plugin-plugins 3.1.2 (core)",
    "@oclif/plugin-search 0.0.17 (core)",
    "@oclif/plugin-update 3.1.17 (core)",
    "@oclif/plugin-version 1.3.4 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.37 (core)",
    "@oclif/plugin-which 2.2.22 (core)",
    "@salesforce/cli 1.83.0 (core)",
    "apex 2.3.3 (core)",
    "auth 2.8.1 (core)",
    "data 2.3.22 (core)",
    "deploy-retrieve 1.11.7 (core)",
    "info 2.6.19 (core)",
    "limits 2.3.18 (core)",
    "login 1.2.11 (core)",
    "org 2.9.8 (core)",
    "schema 2.3.12 (core)",
    "settings 1.4.11 (core)",
    "sobject 0.1.24 (core)",
    "source 2.10.14 (core)",
    "telemetry 2.2.0 (core)",
    "templates 55.4.20 (core)",
    "trust 2.4.20 (core)",
    "user 2.3.17 (core)",
    "@salesforce/sfdx-scanner 3.12.0 (user)"
  ]
}
```

**Comment by WillieRuemmele** [comment]
This has been released in at least `@salesforce/cli/2.1.7 darwin-arm64 node-v18.16.0`
```bash
 ➜  sf project deploy start --source-dir force-app/main --async
Deploying v58.0 metadata to test-t8b6uypyb42i@example.com using the v58.0 SOAP API.
Deploy ID: 0AfDR00002imMvq0AE
Deploy has been queued.

Run "sf project deploy resume --job-id 0AfDR00002imMvq0AE" to resume watching the deploy.
Run "sf project deploy report --job-id 0AfDR00002imMvq0AE" to get the latest status.
Run "sf project deploy cancel --job-id 0AfDR00002imMvq0AE" to cancel the deploy.
➜  dreamhouse-lwc git:(main) ✗  hub:(DevHub) scratch:(test-t8b6uypyb42i@example.com)
 ➜  cd ../           
➜  scratches 
 ➜  sf project deploy cancel --use-most-recent                 
Successfully canceled 0AfDR00002imMvq0AE
```

**Comment by hudec117** [comment]
There are other commands which will not need to be executed from a project, for example the `sf org list metadata-types` command.

Instead of waiting for people to raise an issue for each one, it would be nice to see each command reviewed and the project requirement removed if it doesn't depend on any project-specific information/files.

I don't see an issue with this for commands that only need to know the target environment.
